### ISSUE TITLE
IDE0054 doesn't trigger for null coalescing assignment

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#38059]
**Version Used**: 
3.3.0-beta3-19413-06+ac06df1bffb7b7fd0e5bf63cc91921af76b21e0

**Steps to Reproduce**:
```C#
using System;

class Program
{
    static void Main() => Console.WriteLine(Foo);

    private static string s_foo;
    private static string Foo => s_foo ?? (s_foo = new string('c', 42));
}
``` 

**Expected Behavior**:
I was expecting that IDE0054 (or some other rule) would trigger to translate this into:
```C#
using System;

class Program
{
    static void Main() => Console.WriteLine(Foo);

    private static string s_foo;
    private static string Foo => s_foo ??= new string('c', 42);
}
```

**Actual Behavior**:
No refactoring was available.

Maybe this is a feature request rather than a bug?
